Assets/Scripts/Model/Weapon.cs

```diff
@@ -98,7 +105,7 @@ public CanReceiveDamage getAvailableTarget()
             var target = targets[0];
 
             // Check if target is already dead
-            if (target.Equals(null))
+			if (target.Equals(null) || target == null)
```

Comment:
Apparently not, I got missing references sometimes and I investigated, this is the difference:
"The first condition prevents situation when the target really becomes null, while second one correctly uses Equals overriden in UnityEngine.Object."
I made this change along with other changes ... so I don't really know if this solved the error or something else did.